.github/workflows/release.yaml

```diff
@@ -228,6 +228,9 @@ jobs:
       - name: Merge with main
         run: git merge origin/main --no-ff -Xtheirs -m "Merge branch 'main' into temporary docs branch"
 
+      - name: Rebase onto docs-deploy
+        run: git rebase docs-deploy -Xours
```

Comment:
The key here is to use the "ours" merge strategy here to take any changes from `main` and override the changes in the `docs-deploy` branch whenever there are conflicts.